boot/bootutil/src/loader.c

```diff
@@ -2214,17 +2229,22 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
     if(FIH_NOT_EQ(fih_cnt, BOOT_IMAGE_NUMBER)) {
         FIH_PANIC;
     }
+
+    fill_rsp(state, rsp);
+
+    fih_rc = FIH_SUCCESS;
+out:
     /*
      * Since the boot_status struct stores plaintext encryption keys, reset
      * them here to avoid the possibility of jumping into an image that could
      * easily recover them.
      */
+#if defined(MCUBOOT_ENC_IMAGES) || defined(MCUBOOT_SWAP_SAVE_ENCTLV)
+    like_mbedtls_zeroize(&bs, sizeof(bs));
```

Comment:
I think you should also test if Mbed-TLS is being used, and in that case call `mbedtls_platform_zeroize`. Otherwise just use `memset` like below. If I remember correctly Tinycrypt uses `memset` without any "volatile" tricks or anything, so I assume it will also leave behind its state, which includes the encryption keys. I have to double check this, if my setup also gets rid of the `memset` calls! Apart from this PR is good, but it's just a suggestion anyway...

Reply:
Which means `like_mbedtls_zeroize` can be removed, or it should be something like `bootutil_zeroize` but then we have to use it in all places where it's required, including the Tinycrypt code.